### PR TITLE
ci: add label when user respond in the pr

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

Add a `user respond` label when someone comment in the pr, it's very useful for pr reviewers.
I tested in the pr: https://github.com/monkeyDluffy6017/apisix/pull/6

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
test

**Comment by monkeyDluffy6017** [comment]
test

File: .github/workflows/update-labels.yml [hunk 1]
```diff
@@ -28,3 +30,33 @@ jobs:
...
+          script: |
+            github.rest.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ["user responded"]
+            })
+
+  pr_review_commented:
+    if: github.event.pull_request && github.event.comment.user.login == github.event.pull_request.user.login && (contains(github.event.pull_request.labels.*.name, 'wait for update') || contains(github.event.pull_request.labels.*.name, 'discuss')) && !contains(github.event.pull_request.labels.*.name, 'user responded')
```
Comment (shreemaan-abhishek) [inline .github/workflows/update-labels.yml]: I think it should be:
```suggestion
    if: github.event.pull_request && github.event.pull_request_review_comment.user.login == github.event.pull_request.user.login && (contains(github.event.pull_request.labels.*.name, 'wait for update') || contains(github.event.pull_request.labels.*.name, 'discuss')) && !contains(github.event.pull_request.labels.*.name, 'user responded')
```

https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request_review_comment

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]
Are you sure you do not want to do the automation when the user pushes a new commit?

File: .github/workflows/update-labels.yml (hunk 1 above)
Comment (monkeyDluffy6017) [reply .github/workflows/update-labels.yml]: The `github.event` stands for the event `pull_request_review_comment`, so we just need to get the property `comment` from `github.event`

Comment (shreemaan-abhishek) [reply .github/workflows/update-labels.yml]: I hope you performed a manual test/demo in a dummy repository. It's a good idea to do so.

**Comment by monkeyDluffy6017** [comment]
> Are you sure you do not want to do the automation when the user pushes a new commit?

@shreemaan-abhishek I can't get the label value from the `push` event, so i think it's fine to focus on the comment, the author need to notify us when he is ready

File: .github/workflows/update-labels.yml (hunk 1 above)
Comment (monkeyDluffy6017) [reply .github/workflows/update-labels.yml]: Yeah, i metioned this in the description
```
I tested in the pr: https://github.com/monkeyDluffy6017/apisix/pull/6
```

**Review by leslie-tsang — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

File: .github/workflows/update-labels.yml [hunk 2]
```diff
@@ -28,3 +30,33 @@ jobs:
...
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ["user responded"]
+            })
+
```
Comment (shreemaan-abhishek) [inline .github/workflows/update-labels.yml:48]: `removeLabel` to remove "wait for update" label is missing. 😬 

ref:
https://github.com/monkeyDluffy6017/apisix/blob/d643de37c2701861fbcf4775d0dcc0d049954fb9/.github/workflows/update-labels.yml#L27-L32

**Pull request merged**